src/frontend/src/tables/part/PartTable.tsx

```diff
@@ -380,10 +449,17 @@ export function PartListTable({
             setSelectedPart(record.pk);
             editPart.open();
           }
+        }),
+        RowDuplicateAction({
+          hidden: !can_add,
+          onClick: () => {
+            setSelectedPart(record.pk);
```

Comment:
You can store the provided `record` instance here, rather than fetching it again